misk-crypto/src/main/kotlin/misk/crypto/Cipher.kt

```diff
@@ -0,0 +1,69 @@
+package misk.crypto
+
+import com.google.crypto.tink.Aead
+import com.google.crypto.tink.JsonKeysetWriter
+import com.google.crypto.tink.KeysetHandle
+import com.google.crypto.tink.aead.AeadFactory
+import okio.ByteString
+import okio.ByteString.Companion.toByteString
+import java.io.ByteArrayOutputStream
+import java.util.Base64
+
+/**
+ * An even simpler interface than Tink's [Aead].
+ * This also makes it easier to mock Ciphers for testing purposes.
+ */
+interface Cipher {
+  /**
+   * Encrypt the given data.
+   * If anything goes wrong, throws a [java.security.GeneralSecurityException]
+   */
+  fun encrypt(plaintext: ByteString): ByteString
+
+  /**
+   * Decrypt the given data
+   * If anything goes wrong, including failing to decipher the data,
+   * throws a [java.security.GeneralSecurityException]
```

Comment:
I think you might want a different behavior for decryption failure? Maybe a better exception or it could return a `ByteString?`

Reply:
The interface Google provides in Tink specifically says that these methods may throw a `GeneralSecurityException`.
I think it'd be a good approach to stick with those exceptions. hiding/swallowing these exceptions could be problematic and could even lead to data corruption.
I think that if the app fails to encrypt/decrypt anything, it's either going to be something widespread (like a bad key was setup somehow), or as you said, probably won't happen at all.
In any case, I think you want the app to blow-up in these cases.